build-tools/plots-to-appinsights/AppInsights.cs

```diff
@@ -0,0 +1,39 @@
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using System;
+using System.Collections.Generic;
+
+namespace ProcessPlotCSVFile
+{
+	internal class AppInsights
+	{
+		private TelemetryClient TelemetryClient { get; set; } = null;
+
+		public AppInsights (string appInsightsInstrumentationKey)
+		{
+			if (string.IsNullOrEmpty (appInsightsInstrumentationKey)) {
+				throw new ArgumentNullException (nameof (appInsightsInstrumentationKey));
+			}
+
+			// https://docs.microsoft.com/en-us/azure/azure-functions/functions-monitoring#log-custom-telemetry-in-c-functions
+			var telemetryConfiguration = new TelemetryConfiguration (appInsightsInstrumentationKey);
+			TelemetryClient = new TelemetryClient (telemetryConfiguration)  // Note: Direct creation of the TelemetryClient using the default constructor has been deprecated. Use dependency injection instead.
+			{
+				InstrumentationKey = appInsightsInstrumentationKey
+			};
+		}
+
+		public void SendTelemetry (string eventName, Dictionary<string, string> properties, Dictionary<string, double> metrics)
+		{
+			if (TelemetryClient != null && !string.IsNullOrWhiteSpace (TelemetryClient.InstrumentationKey)) {
+				lock (TelemetryClient) {
+					TelemetryClient.TrackEvent (eventName, properties, metrics);
+				}
+			} else {
+				throw new Exception ("TelemetryClient not initialized or AppInsights instrumentation key not set");
```

Comment:
Is it possible for `TelemetryClient` to be `null`?  It shouldn't be, and thus this `else` block can be removed.

Reply:
Indeed the null/empty checks aren't needed as the constructor should keep everything honest.
Addressed with commit: 135324e3afd3da9c3c91a9de110ae83e56df3353